components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/model/Authentication.java

```diff
@@ -233,4 +236,60 @@ public Authentication build() {
             return new Authentication(this);
         }
     }
+
+    /**
+     * This builder build endpoint by taking the authentication type and properties as input.
+     */
+    public static class AuthenticationBuilder {
+
+        private Type authType;
+        private Map<String, String> authPropertiesMap;
+
+        public AuthenticationBuilder type(Type type) {
+
+            this.authType = type;
+            return this;
+        }
+
+        public AuthenticationBuilder properties(Map<String, String> authPropertiesMap) {
+
+            this.authPropertiesMap = authPropertiesMap;
+            return this;
+        }
+
+        public Authentication build() {
+
+            switch (authType) {
+                case BASIC:
+                    return new Authentication.BasicAuthBuilder(
+                            getProperty(authPropertiesMap, Property.USERNAME.getName()),
+                            getProperty(authPropertiesMap, Property.PASSWORD.getName())).build();
+                case BEARER:
+                    return new Authentication.BearerAuthBuilder(
+                            getProperty(authPropertiesMap, Property.ACCESS_TOKEN.getName())).build();
+                case API_KEY:
+                    return new Authentication.APIKeyAuthBuilder(
+                            getProperty(authPropertiesMap, Property.HEADER.getName()),
+                            getProperty(authPropertiesMap, Property.VALUE.getName())).build();
+                case NONE:
+                    return new Authentication.NoneAuthBuilder().build();
+                default:
+                    throw new IllegalArgumentException();
```

Comment:
We can give a message for this exception right which mentions why it is an illegal argument